src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -0,0 +1,517 @@
+#include <gtest/gtest.h>
+
+#include "extlibs/hrvo/simulator.h"
+#include "shared/2021_robot_constants.h"
+
+class TestHrvo : public testing::Test
+{
+};
+
+void assertRobotInAgentList(const Robot &robot, const AgentType &agent_type,
+                            const std::list<std::shared_ptr<Agent>> &agents)
+{
+    auto agent_it = std::find_if(agents.begin(), agents.end(),
+                                 [&robot, &agent_type](std::shared_ptr<Agent> agent) {
+                                     return (agent->getRobotId() == robot.id() &&
+                                             agent->getAgentType() == agent_type);
+                                 });
+    ASSERT_TRUE(agent_it != agents.end());
+}
+
+TEST_F(TestHrvo, test_update_world_empty_agents)
+{
+    HRVOSimulator sim =
+        HRVOSimulator(200u, create2021RobotConstants(), TeamColour::YELLOW);
+    World world = World(Field::createSSLDivisionBField(),
+                        Ball(Point(0, 0), Vector(0, 0), Timestamp()), Team(), Team());
+
+    sim.updateWorld(world);
+
+    EXPECT_EQ(0, sim.getNumAgents());
+}
+
+TEST_F(TestHrvo, test_update_world_with_one_friendly_agent)
+{
+    Robot friendly_robot =
+        Robot(0, Point(), Vector(), Angle(), AngularVelocity(), Timestamp());
+    HRVOSimulator sim =
+        HRVOSimulator(200u, create2021RobotConstants(), TeamColour::YELLOW);
+    World world = World(Field::createSSLDivisionBField(),
+                        Ball(Point(0, 0), Vector(0, 0), Timestamp()),
+                        Team({friendly_robot}), Team());
```

Comment:
nit: To make the tests shorter and easier to read, many of the duplicated variables can be defined once in the class definition. This could probably include the robots, hrvo sim, and the world.

Reply:
I did this, but it didn't really help because each test has a different number of robots in the Team, thus a different world